general/chainerrl/baselines/action_wrappers.py

```diff
@@ -0,0 +1,340 @@
+        If specified, the "noop" action is also affected.
+    exclude_keys
+        List of action keys, which should be ignored for discretizing action space.
+    exclude_noop
+        The "noop" will be excluded from discrete action list.
+    allow_pitch
+        If it is true, it enables a converter to take pitch control action.
+    num_camera_discretize
+        Number of discretization of yaw control (must be odd).
+    max_camera_range
+        Maximum value of yaw control.
+
+    '''
+    env_name = env_name.replace("Dense", "")
+
+    prioritized_elements = [] if prioritized_elements is None else prioritized_elements
+    always_keys = [] if always_keys is None else always_keys
+    reverse_keys = [] if reverse_keys is None else reverse_keys
+    exclude_keys = [] if exclude_keys is None else exclude_keys
+
+    # Set indices on discrete actions
+    actions = []
+    for key in BINARY_KEYS + ['camera'] + list(NUM_ENUM_ACTIONS[env_name]):
+        if (key not in (always_keys + exclude_keys)):
+            actions.append(key)
+
+    discrete_indices = generate_mapping(
+        env_name, actions, num_camera_discretize, allow_pitch, exclude_noop)
+
+    # Set priorities
+    _original_order = [
+        'nearbyCraft', 'nearbySmelt', 'craft', 'equip', 'place', 'camera',
+        'forward', 'back', 'left', 'right', 'jump', 'sneak', 'sprint',
+        'attack']
+    original_order = []
+    for key in _original_order:
+        if key in actions:
+            original_order.append(key)
+
+    # sanity check
+    for key in prioritized_elements:
+        if key not in actions:
+            raise ValueError('Unknown key {} in prioritized_elements.'.format(key))
+
+    priorities = generate_priority(original_order, prioritized_elements)
+
+    # sanity check
+    for key in priorities:
+        if key not in actions:
+            raise ValueError('Unknown key {} in priorities.'.format(key))
+
+    # Set query
+    query = generate_query(
+        env_name, priorities, reverse_keys, discrete_indices,
+        max_camera_range, num_camera_discretize, allow_pitch)
+
+    return DiscreteActionConverter(query)
+
+
+def generate_continuous_converter(env_name, allow_pitch, max_camera_range):
+    '''
+    Generate a continuous action converter for expert dataset.
+
+    Parameters
+    ----------
+    env_name
+        Environment name.
+    allow_pitch
+        If it is true, it enables a converter to take pitch control action.
+    max_camera_range
+        Maximum value of yaw control.
+    '''
+
+    env_name = env_name.replace("Dense", "")
+
+    def action_converter(action):
+        episode_len = len(action['forward'])
+        value = np.zeros((episode_len, len(action) + 1), dtype=np.float32)
+        idx = 0
+        for key in ALL_ORDERED_KEYS:
+            if key not in action:
+                continue
+            if key in BINARY_KEYS:
+                value[:, idx] = -1 + action[key] * 2
```

Comment:
if you're using 0.5 for splitting the binary keys, shouldn't they be mapped to {0,1} instead of {-1,1}

Reply:
The main reason of mapping to {-1, 1} is because tanh is used at `continuous` model's output.
But in this case, it should be divided by 0.0 separator.
I'll fix it. Thanks!